File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/GetEnvironmentsByProjectExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class GetEnvironmentsByProjectExample
{
    fun getEnvironmentsByProject()
    {
        ApiClient.apiKey["Authorization"] = "YOUR_API_KEY"

        try
        {
            val response = EnvironmentsApi().getEnvironmentsByProject(
                projectKey = null,
                limit = null,
                offset = null,
                filter = null,
                sort = null,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling EnvironmentsApi#getEnvironmentsByProject")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling EnvironmentsApi#getEnvironmentsByProject")
            e.printStackTrace()
        }
    }
}

File: launchdarkly/typescript-node/src/GetEnvironmentsByProjectExample.ts
import * as fs from 'fs';
import api from "launchdarkly_client"
import models from "launchdarkly_client"

const apiCaller = new api.EnvironmentsApi();
apiCaller.setApiKey(api.EnvironmentsApiApiKeys.ApiKey, "YOUR_API_KEY");

apiCaller.getEnvironmentsByProject(
  "projectKey_string", // projectKey
  undefined, // limit
  undefined, // offset
  undefined, // filter
  undefined, // sort
).then(response => {
  console.log(response.body);
}).catch(error => {
  console.log("Exception when calling EnvironmentsApi#getEnvironmentsByProject:");
  console.log(error.body);
});
